scripts/prepare-template.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+PACKAGE_LOCATION=$(pwd)/react-native-$1.tgz
+
+# Pack React Native into a `.tgz` file so we can run from source
+npm pack
+
+# Set the React Native version to point to the `.tgz` file
+node scripts/set-rn-template-version.js "file:$PACKAGE_LOCATION"
+
+# We need to generate CocoaPods project. To do so, we install depdendencies
+# locally and manually run `pod install`
+cd template
+
+npm install
```

Comment:
`npm install` creates `package-lock.json` (it's in gitignore so you don't see it). Can we use Yarn instead (it's faster) and checkout `yarn.lock` along with `package.json`?

Reply:
I didn't go with `yarn` for two reasons:
- it has issues with cache when installing from a tarball - it will quickly eat up available space unless you `yarn cache clean` yourself
- we use `npm install` and `npm pack` already in other `scripts/*`, so figured I'll just stick to the pattern